src/app/api/core/utils/withRetry.ts

```diff
@@ -0,0 +1,49 @@
+import { StatusableError } from '@/types/CopilotApiError'
+import pRetry, { FailedAttemptError } from 'p-retry'
+import * as Sentry from '@sentry/nextjs'
+
+export const withRetry = async <T>(fn: (...args: any[]) => Promise<T>, args: any[]): Promise<T> => {
+  let isEventProcessorRegisterd = false
+
+  return await pRetry(
+    async () => {
+      try {
+        return await fn(...args)
+      } catch (error) {
+        // Hopefully now sentry doesn't report retry errors as well. We have enough triage issues as it is
+        Sentry.withScope((scope) => {
+          if (isEventProcessorRegisterd) return
+
+          isEventProcessorRegisterd = true
+          scope.addEventProcessor((event) => {
+            if (event.level === 'error' && event.message && event.message.includes('An error occurred during retry')) {
+              return null // Discard the event as it occured during retry
```

Comment:
Here we are relying on `event.message` including "An error occurred during retry" string, which can be fragile if error messages change in the future. Can we consider adding a more explicit marker to the error or checking for more reliable properties, like a specific error code or type, to prevent accidentally filtering out unrelated errors?